diff --git a/node_modules/@instructure/ui-truncate-text/es/TruncateText/utils/truncate.js b/node_modules/@instructure/ui-truncate-text/es/TruncateText/utils/truncate.js
index a8c7d58..a6b229f 100644
--- a/node_modules/@instructure/ui-truncate-text/es/TruncateText/utils/truncate.js
+++ b/node_modules/@instructure/ui-truncate-text/es/TruncateText/utils/truncate.js
@@ -177,10 +177,26 @@ class Truncator {
     return html;
   }
   checkFit(data) {
-    const html = this.domString(data);
     const node = this._options.maxLines === 'auto' ? this._stage : this._parent;
     let fits = true;
-    this._stage.innerHTML = html;
+    while (this._stage.firstChild) {
+      this._stage.removeChild(this._stage.firstChild);
+    }
+    for (let i = 0; i < data.length; i++) {
+      const mapItem = this._nodeMap[i];
+      const text = data[i].join('');
+      if (mapItem.node.nodeType === Node.ELEMENT_NODE) {
+        const el = document.createElement(mapItem.node.nodeName);
+        const attr = mapItem.node.attributes;
+        for (let j = 0; j < attr.length; j++) {
+          el.setAttribute(attr[j].nodeName, attr[j].nodeValue);
+        }
+        el.textContent = text;
+        this._stage.appendChild(el);
+      } else if (mapItem.node.nodeType === 3) {
+        this._stage.appendChild(document.createTextNode(text));
+      }
+    }
     // allow a 0.5 px margin of error for browser calculation discrepancies
     if (getBoundingClientRect(node).height - this._maxHeight > 0.5) {
       fits = false;
diff --git a/node_modules/@instructure/ui-truncate-text/lib/TruncateText/utils/truncate.js b/node_modules/@instructure/ui-truncate-text/lib/TruncateText/utils/truncate.js
index 6deb59e..5ed6492 100644
--- a/node_modules/@instructure/ui-truncate-text/lib/TruncateText/utils/truncate.js
+++ b/node_modules/@instructure/ui-truncate-text/lib/TruncateText/utils/truncate.js
@@ -186,10 +186,26 @@ class Truncator {
     return html;
   }
   checkFit(data) {
-    const html = this.domString(data);
     const node = this._options.maxLines === 'auto' ? this._stage : this._parent;
     let fits = true;
-    this._stage.innerHTML = html;
+    while (this._stage.firstChild) {
+      this._stage.removeChild(this._stage.firstChild);
+    }
+    for (let i = 0; i < data.length; i++) {
+      const mapItem = this._nodeMap[i];
+      const text = data[i].join('');
+      if (mapItem.node.nodeType === Node.ELEMENT_NODE) {
+        const el = document.createElement(mapItem.node.nodeName);
+        const attr = mapItem.node.attributes;
+        for (let j = 0; j < attr.length; j++) {
+          el.setAttribute(attr[j].nodeName, attr[j].nodeValue);
+        }
+        el.textContent = text;
+        this._stage.appendChild(el);
+      } else if (mapItem.node.nodeType === 3) {
+        this._stage.appendChild(document.createTextNode(text));
+      }
+    }
     // allow a 0.5 px margin of error for browser calculation discrepancies
     if ((0, _getBoundingClientRect.getBoundingClientRect)(node).height - this._maxHeight > 0.5) {
       fits = false;
diff --git a/node_modules/@instructure/ui-truncate-text/src/TruncateText/utils/truncate.ts b/node_modules/@instructure/ui-truncate-text/src/TruncateText/utils/truncate.ts
index f8ed936..a7c074d 100644
--- a/node_modules/@instructure/ui-truncate-text/src/TruncateText/utils/truncate.ts
+++ b/node_modules/@instructure/ui-truncate-text/src/TruncateText/utils/truncate.ts
@@ -235,10 +235,26 @@ class Truncator {
   }
 
   checkFit(data: string[][]) {
-    const html = this.domString(data)
     const node = this._options.maxLines === 'auto' ? this._stage : this._parent
     let fits = true
-    this._stage.innerHTML = html
+    while (this._stage.firstChild) {
+      this._stage.removeChild(this._stage.firstChild)
+    }
+    for (let i = 0; i < data.length; i++) {
+      const mapItem = this._nodeMap[i]
+      const text = data[i].join('')
+      if (mapItem.node.nodeType === Node.ELEMENT_NODE) {
+        const el = document.createElement(mapItem.node.nodeName)
+        const attr = (mapItem.node as Element).attributes
+        for (let j = 0; j < attr.length; j++) {
+          el.setAttribute(attr[j].nodeName, attr[j].nodeValue!)
+        }
+        el.textContent = text
+        this._stage.appendChild(el)
+      } else if (mapItem.node.nodeType === 3) {
+        this._stage.appendChild(document.createTextNode(text))
+      }
+    }
     // allow a 0.5 px margin of error for browser calculation discrepancies
     if (getBoundingClientRect(node).height - this._maxHeight > 0.5) {
       fits = false
